commit b4250dd868d1b42c0a65de11ef3afbee67ba5d2f
Author: Olga Kornievskaia <kolga@netapp.com>
Date:   Thu Mar 11 10:55:00 2021 -0500

    NFSD: fix error handling in NFSv4.0 callbacks
    
    When the server tries to do a callback and a client fails it due to
    authentication problems, we need the server to set callback down
    flag in RENEW so that client can recover.
    
    Suggested-by: Bruce Fields <bfields@redhat.com>
    Signed-off-by: Olga Kornievskaia <kolga@netapp.com>
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
    Tested-by: Benjamin Coddington <bcodding@redhat.com>
    Link: https://lore.kernel.org/linux-nfs/FB84E90A-1A03-48B3-8BF7-D9D10AC2C9FE@oracle.com/T/#t

diff --git a/fs/nfsd/nfs4callback.c b/fs/nfsd/nfs4callback.c
index 052be5bf9ef5..7325592b456e 100644
--- a/fs/nfsd/nfs4callback.c
+++ b/fs/nfsd/nfs4callback.c
@@ -1189,6 +1189,7 @@ static void nfsd4_cb_done(struct rpc_task *task, void *calldata)
 		switch (task->tk_status) {
 		case -EIO:
 		case -ETIMEDOUT:
+		case -EACCES:
 			nfsd4_mark_cb_down(clp, task->tk_status);
 		}
 		break;